### PR TITLE
[FindNextAction] Introduce unit tests

### PR DESCRIPTION
**Pull request by Wittmaxi**
Introduces unit tests for the FindNextAction.
I don't achieve 100% test coverage, but I test for the most important parts. Appart from the "naive" functionality, in particular I test for:
1) search options are taken into account
2) the search history is honored
3) whole - word search is only performed if the selection is a word
4) RegExes are correctly escaped

### CODE REVIEW

**Comment by Wittmaxi** [comment]
The test is now in the editors.test package. Reason is that I need to start a whole editor in order to test the FindNextAction using IDE.openEditor. Depending on IDE in the workbench.editors package is problematic and such @HeikoKlare and I convened on moving the test to a more sensible place.

File: tests/org.eclipse.ui.editors.tests/src/org/eclipse/ui/editors/tests/FindNextActionTest.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+	private static ResourceBundle fgBundleForConstructedKeys= ResourceBundle.getBundle(BUNDLE_FOR_CONSTRUCTED_KEYS);
+
+	public void openEditorAndFindNextAction(String content, boolean direction) {
+		try {
+			IFolder folder= ResourceHelper.createFolder("EncodingChangeTestProject/EncodingChangeTests/");
+			fFile= ResourceHelper.createFile(folder, "file" + fCount + ".txt", content);
+			fFile.setCharset(null, null);
+			fCount++;
+		} catch (CoreException e) {
+			fail();
```
Comment (akurtakov) [inline tests/org.eclipse.ui.editors.tests/src/org/eclipse/ui/editors/tests/FindNextActionTest.java]: Please no such empty fail() calls, it becomes a nightmare figuring out what happened when such a fail occurs. As a bare minimum pass e.getMessage() but it's even better to not catch the exception but let the method throws it and let JUnit fail the test itself with the message and full stack.

File: tests/org.eclipse.ui.editors.tests/src/org/eclipse/ui/editors/tests/FindNextActionTest.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+		}
+		IWorkbench workbench= PlatformUI.getWorkbench();
+		IWorkbenchPage page= workbench.getActiveWorkbenchWindow().getActivePage();
+		try {
+			fEditor= IDE.openEditor(page, fFile);
+		} catch (PartInitException e) {
+			fail();
```
Comment (akurtakov) [inline tests/org.eclipse.ui.editors.tests/src/org/eclipse/ui/editors/tests/FindNextActionTest.java]: Likewise.

File: tests/org.eclipse.ui.editors.tests/src/org/eclipse/ui/editors/tests/FindNextActionTest.java (hunk 1 above)
Comment (HeikoKlare) [reply tests/org.eclipse.ui.editors.tests/src/org/eclipse/ui/editors/tests/FindNextActionTest.java]: Yes, these fail statements are actually completely unnecessary. I am currently updating the PR accordingly (and improving it in general).

**Comment by Wittmaxi** [comment]
@HeikoKlare thank you for updating the PR. I had copied over the method for opening an editor from another test - yet I agree that failing siletnyl is not a good idea.

**Comment by HeikoKlare** [comment]
Thank you for adding the tests, @Wittmaxi! They are executed and succeed as expected.
Other failing tests are unrelated and documented: #751, #370

**Pull request merged**